### PR TITLE
host: Add realm writability icon in file tree

### PR DESCRIPTION
**Pull request by backspace**
There’s no final design for this but here’s a placeholder:

Boxel Tests 2024-02-12 14-00-27

### CODE REVIEW

**Comment by backspace** [comment]
One of the Percy diffs:

Build #3410 - Cardstack - Percy | Visual testing as a service 2024-02-12 14-26-51

This would already have been a problem with longer names but the additional icon exacerbates it. Do we think wrapping is desirable or should I add truncation like [here](https://github.com/cardstack/boxel/pull/1024) so they stay on one line?

**Comment by burieberry** [comment]
> This would already have been a problem with longer names but the additional icon exacerbates it. Do we think wrapping is desirable or should I add truncation like [here](https://github.com/cardstack/boxel/pull/1024) so they stay on one line?

I'm fine with truncating it since it's possible to expand the panel. Another option is truncating after 2 lines. Here it also looks like percy has some additional padding on the left of the file tree panel for some reason.

**Review by burieberry — COMMENTED** [review]
Hmm I wonder if people who don't have edit permission should even be in code mode. But if this is what was decided, the changes look good.

File: packages/host/app/components/editor/file-tree.gts [hunk 1]
```diff
@@ -71,19 +105,31 @@ export default class FileTree extends Component<Signature> {
...
 
   @service private declare router: RouterService;
   @tracked private showMask = true;
+
+  private realmResource: RealmResource | undefined;
+
   constructor(owner: Owner, args: Signature['Args']) {
     super(owner, args);
     this.hideMask.perform();
+    this.realmResource = getRealm(this, () => this.args.realmURL);
```
Comment (habdelra) [inline packages/host/app/components/editor/file-tree.gts]: just FYI, the API for the realm resource changed a little bit in a recent PR. after you merge upstream, this now looks like:
```ts
    this.realmResource = getRealm(this, { realmURL: () => this.args.realmURL });
```

**Review by habdelra — APPROVED** [review]

**Comment by habdelra** [comment]
> Hmm I wonder if people who don't have edit permission should even be in code mode. But if this is what was decided, the changes look good.

i think it still makes sense. keep in mind that the base realm will be read only--i think that it helps developers to understand how a card is implemented to read the code for a card, even if they can't change it.

File: packages/host/app/components/editor/file-tree.gts (hunk 1 above)
Comment (backspace) [reply packages/host/app/components/editor/file-tree.gts]: thanks, updated now!

**Comment by backspace** [comment]
> > Hmm I wonder if people who don't have edit permission should even be in code mode. But if this is what was decided, the changes look good.
> 
> i think it still makes sense. keep in mind that the base realm will be read only--i think that it helps developers to understand how a card is implemented to read the code for a card, even if they can't change it.

ya my next task is making the editor read-only when appropriate, agreed that it’s helpful to be able to see the code regardless of edit permissions

re truncation I’ve added it now:

screencast 2024-02-13 11-02-17

**Pull request merged**